Replace validation logic for data-dir in pt-osc to use innodb_directories.

peppla opened the pull request:
- [X] The contributed code is licensed under GPL v2.0
- [X] Contributor Licence Agreement (CLA) is signed
- [ ] util/update-modules has been ran
- [ ] Documentation updated
- [ ] Test suite update

Current validation checks that the directory exists and is writable by the user running pt-osc. This does not make sense because the tool can be executed remotely, and even locally, access is not required for the user running the tool. What is required is that the directory exists in `innodb_directories`.

This change removes the old validation and checks if the directory passed in data-dir exists.

peppla committed:
Replaced validation logic for data-dir

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=1023) <br/>All committers have signed the CLA.

svetasmirnova reviewed (changes requested):
Thank you for the contribution. Please add a test case as described in CONTRIBUTING.md

svetasmirnova commented:
It is also possible to create such tables in innodb_data_home_dir.

svetasmirnova committed:
PR-1023 - Replace validation logic for data-dir in pt-osc to use innodb_directories.

- Data directory could be in innodb_data_home_dir, added this case as well
- Added test case
- Removed diagnostic messages from t/pt-online-schema-change/pt-244.t

svetasmirnova commented:
@peppla I added test case myself and also added a check in case if the directory is innodb_data_home_dir (also allowed). Please check changes and let me know if you disagree with them.

peppla commented:
I agree, of course. Thank you very much!

saikumar-vs commented on bin/pt-online-schema-change:
```diff
@@ -11438,13 +11440,37 @@ sub create_new_table {
+         } split(/\s*;\s*/, $innodb_dirs);
+         my %dirs_lookup = map { $_ => 1 } @dirs;
+         my $data_dir = $o->get('data-dir');
+         $data_dir =~ s{(?<!^)/\z}{};
```
This can be removed,  trailing-slash regex applied again.
